diagnostics/include/diagnostics_server.h

```diff
@@ -133,6 +104,22 @@ inline void handleRequest(Registrar& registrar, int sock) {
 // async connections via boost ASIO if necessary, although this seems extremely heavy handed for the
 // use case.
 class Server {
+  bool setTimeOut(int fd, time_t seconds = 1) {
```

Comment:
Nit: `chrono::seconds` ?

Reply:
In order to set socket timeout and preserve the logic as it was before (handling 1 Client Request at a time) we use **setsockopt**. This is a Linux interface call, which takes a structure **timeval**. In this structure the seconds are represented as **time_t**.